### ISSUE TITLE
Google & Facebook oauth sign up and sign in

### ISSUE THREAD

**Issue by jaiakash** [CCExtractor/beacon#160]
**Is your feature request related to a problem? Please describe.**
Currently, the app only accepts email sign in/signup. Having oauth would be better for the user, then user can sign in with ease.

**Describe the solution you'd like**
Having a google and/or Facebook oauth at sign-in page of app.

**Additional context**

**Are you working on this? (Yes/No)**
Not right now, but I get confirmation from maintainer, I would start work on it.

**Comment by codewithkushagra** [comment]
I would like to work on this, I have implemented oauth in flutter previously, I can do it 
Can you assign it to me

**Comment by nb9960** [comment]
@jaiakash @codewithkushagra Yes, this was planned but how will you guys implement it without updating the backend?

**Comment by codewithkushagra** [comment]
Actually, we don't have to make too many changes to backend, as by using google_sigin package we can maintain the login states, access tokens and refresh token in frontend itself and I will try to make the required changes in backend.

**Comment by jaiakash** [comment]
Yeah i agree, there are packages for facebook and google oauth which we can use without much changes in backend.
Blog link - [link]( https://fayaz07.medium.com/social-authentication-facebook-and-google-in-flutter-without-firebase-e3ca289ed50c)
[Facebook package](https://pub.dev/packages/flutter_facebook_auth)
[Google package](https://pub.dev/packages/google_sign_in)

**Comment by ItsAdityaKSingh** [comment]
Are you working on proposing changes for this issue @jaiakash?

**Comment by codewithkushagra** [comment]
Hey, @ItsAdityaKSingh if @jaiakash is not working on this can I work on it?
I have previously implemented such features so I can implement it easily, let me know if issue open to work on

**Comment by jaiakash** [comment]
> Are you working on proposing changes for this issue @jaiakash?

No, I am not actively working on it. I guess @codewithkushagra can it if he has worked on a similar issue previously.

**Comment by ItsAdityaKSingh** [comment]
Ok! If @codewithkushagra you are willing to resolve this issue, I would like to discuss it first. Like how are you planning to change the backend to implement this feature? Currently, whoever registers, the information regarding the users, list of beacons, and everything is saved in our customized backend, where we easily retrieve the information about them. Using Google or Facebook would need to change how we currently keep records. What do you suggest?

**Comment by asmitranjansinha** [comment]
we can use firebase and google/facebook auth packages

**Comment by asmitranjansinha** [comment]
if anyone is not working on this issue, I can work on it @ItsAdityaKSingh

**Comment by lalitkumawat1m** [comment]
Hi, could you please assign this issue to me?
I would love to work on it and contribute to the project.

**Comment by Kaps61929** [comment]
@ItsAdityaKSingh I would love to contribute to this issue. Please assign it to me.
